CalcualtorLogic.js

```diff
@@ -18,16 +18,91 @@ class CalculatorLogic {
+        resultDiv.appendChild(
+            document.createTextNode(this.currentResult)
+        );
+        this.resultElement = resultDiv;
+        wrapperDiv.appendChild(resultDiv);
+        const row1 = this.getButtonsRow(["7", "8", "9"]);
+        const row2 = this.getButtonsRow(["4", "5", "6"]);
+        const row3 = this.getButtonsRow(["1", "2", "3"]);
+        const row4 = this.getOperatorsRow(["*", "+", "-"]);
+        wrapperDiv.appendChild(row1);
+        wrapperDiv.appendChild(row2);
+        wrapperDiv.appendChild(row3);
+        wrapperDiv.appendChild(row4);
+        const div4 = document.createElement("div");
+        const button0 = document.createElement("button");
+        button0.dataset.number = "0";
+        button0.classList.add("button");
+        const button0Text = document.createTextNode("0");
+        button0.appendChild(button0Text);
+        const periodButton = document.createElement("button");
+        periodButton.classList.add("button");
+        const periodButtonText = document.createTextNode(".");
+        periodButton.appendChild(periodButtonText);
+        const negativeValueButton = document.createElement("button");
+        negativeValueButton.classList.add("button");
+        const negativeValueButtonText = document.createTextNode("±");
+        negativeValueButton.appendChild(negativeValueButtonText);
+        div4.appendChild(button0);
+        div4.appendChild(periodButton);
+        div4.appendChild(negativeValueButton);
+        wrapperDiv.appendChild(div4);
+        this.hostElem.appendChild(wrapperDiv);
+    };
+    getButtonsRow(buttons) {
+        const div = document.createElement("div");
+        for (let button of buttons) {
+            var buttonElement = document.createElement("button");
+            buttonElement.dataset.number = button;
+            buttonElement.classList.add("button");
+            const buttonElementText = document.createTextNode(button);
+            buttonElement.appendChild(buttonElementText);
+            div.appendChild(buttonElement);
+            buttonElement.addEventListener("click", this.onButtonNumberClick);
+        }
+        return div;
+    }
+    getOperatorsRow(buttons) {
+        const div = document.createElement("div");
+        for (let button of buttons) {
+            var buttonElement = document.createElement("button");
+            buttonElement.dataset.operator = button;
+            buttonElement.classList.add("button");
+            const buttonElementText = document.createTextNode(button);
+            buttonElement.appendChild(buttonElementText);
+            div.appendChild(buttonElement);
+        }
+        return div
     };
-    render() {};
+    onButtonNumberClick(event) {
+        const clickedButton = event.target.dataset.number;
+        let currentResult_ = this.currentResult;
+        let clickedButtonAndCurrentResult = currentResult_ + clickedButton;
+        this.currentResult = currentResult_ + clickedButtonAndCurrentResult;
```

Comment:
Can you please explain why you are adding the value of `currentResult_` + `clickedButtonAndCurrentResult` to `this.currentResult` ?

I assume you're doing this for a reason?!

when I click on button 1 for the first time.
`this.currentResult` is 0 and the `clickedButton` is 1
you are adding  0 to 1 = 01 
then you are adding 0 to 01  = 001 

then if I click on button 2
`this.currentResult` is 001 and `clickedButton` is 2
you are adding 001 to 2  = 0012
then you are adding 001 to 0012 = 0010012 
and then you are assigning that value to `this.currentResult`

Reply:
because previously I was guessing the value of calculator is the value of current result 
"3.	add the value of the clicked button to the value of the calculator "
Thanks

Reply:
but you did that here
`let clickedButtonAndCurrentResult = currentResult_ + clickedButton;` 
or did you something else here?

Reply:
my question was why you are adding the `currentResult` twice

Reply:
yes you are right